StyleCop.Analyzers/StyleCop.Analyzers/NamingRules/SA1314ParametersShouldMatchInheritedNames.cs

```diff
@@ -0,0 +1,120 @@
+    /// </remarks>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class SA1314ParametersShouldMatchInheritedNames : DiagnosticAnalyzer
+    {
+        /// <summary>
+        /// The ID for diagnostics produced by the <see cref="SA1314ParametersShouldMatchInheritedNames"/> analyzer.
+        /// </summary>
+        public const string DiagnosticId = "SA1314";
+        private static readonly LocalizableString Title = new LocalizableResourceString(nameof(NamingResources.SA1314Title), NamingResources.ResourceManager, typeof(NamingResources));
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(nameof(NamingResources.SA1314MessageFormat), NamingResources.ResourceManager, typeof(NamingResources));
+        private static readonly LocalizableString Description = new LocalizableResourceString(nameof(NamingResources.SA1314Description), NamingResources.ResourceManager, typeof(NamingResources));
+        private static readonly string HelpLink = "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1314.md";
+
+        private static readonly DiagnosticDescriptor Descriptor =
+            new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, AnalyzerCategory.NamingRules, DiagnosticSeverity.Warning, AnalyzerConstants.EnabledByDefault, Description, HelpLink);
+
+        private static readonly Action<CompilationStartAnalysisContext> CompilationStartAction = HandleCompilationStart;
+        private static readonly Action<SymbolAnalysisContext> HandleMethodAction = HandleMethod;
+
+        /// <inheritdoc/>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(Descriptor);
+
+        /// <inheritdoc/>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(CompilationStartAction);
+        }
+
+        private static void HandleCompilationStart(CompilationStartAnalysisContext context)
+        {
+            context.RegisterSymbolAction(HandleMethodAction, SymbolKind.Method);
+        }
+
+        private static void HandleMethod(SymbolAnalysisContext context)
+        {
+            var symbol = context.Symbol as IMethodSymbol;
+
+            if (!symbol.IsOverride && !NamedTypeHelpers.IsImplementingAnInterfaceMember(symbol))
+            {
+                return;
+            }
+
+            if (!symbol.CanBeReferencedByName
+                || !symbol.Locations.Any(x => x.IsInSource)
+                || string.IsNullOrWhiteSpace(symbol.Name))
+            {
+                return;
+            }
+
+            var pool = SharedPools.Default<List<IMethodSymbol>>();
+
+            using (var pooledObject = pool.GetPooledObject())
+            {
+                var originalDefinitions = pooledObject.Object;
+                NamedTypeHelpers.GetOriginalDefinitions(originalDefinitions, symbol);
```

Comment:
lists are in general larger objects because of the underlying array. This is why I'm using a pool in the analyzer code because it is run on every keystroke possibly often.